api/apps/api/test/integration/typeorm-export.repository.integration.e2e-spec.ts

```diff
@@ -0,0 +1,208 @@
+import { ClonePiece, ComponentId, ResourceKind } from '@marxan/cloning/domain';
+import { Connection } from 'typeorm';
+import { v4 } from 'uuid';
+import { ResourceId } from '../../src/modules/clone/export';
+import { ExportEntity } from '../../src/modules/clone/export/adapters/entities/exports.api.entity';
+import { ComponentLocation } from '../../src/modules/clone/export/application/complete-piece.command';
+import { ExportRepository } from '../../src/modules/clone/export/application/export-repository.port';
+import { Export, ExportId } from '../../src/modules/clone/export/domain';
+import { bootstrapApplication } from '../utils/api-application';
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { ExportComponentEntity } from '@marxan-api/modules/clone/export/adapters/entities/export-components.api.entity';
+import { ComponentLocationEntity } from '@marxan-api/modules/clone/export/adapters/entities/component-locations.api.entity';
+
+describe('Typeorm export repository', () => {
+  let fixtures: FixtureType<typeof getFixtures>;
+
+  beforeEach(async () => {
+    fixtures = await getFixtures();
+  }, 20000);
+
+  afterAll(async () => {
+    await fixtures.cleanup();
+  });
+
+  it('should expose methods for getting an export by id and storing exports', async () => {
+    await fixtures.WhenAnExportIsSavedSuccessfully();
+
+    const exportData = await fixtures.WhenReadingTheSavedExportFromRepository();
+
+    await fixtures.ThenExportDataShouldBeOk({
+      exportData,
+      componentsAreCompleted: false,
+    });
+  });
+
+  it('should update an export when a nested piece is completed', async () => {
+    await fixtures.WhenAnExportIsSavedSuccessfully();
+    await fixtures.WhenAComponentIsCompleted();
+    const exportData = await fixtures.WhenReadingTheSavedExportFromRepository();
+
+    await fixtures.ThenExportDataShouldBeOk({
+      exportData,
+      componentsAreCompleted: true,
+    });
+  });
+
+  it('should save entities working with transaction', async () => {
+    await fixtures.WhenAnExportIsSavedSuccessfullyWithinATransaction();
+    await fixtures.WhenAllComponentsAreCompletedWithTransactions();
+
+    const exportData = await fixtures.WhenReadingTheSavedExportFromRepository();
+
+    fixtures.ThenAllExportComponentsShouldBeFinished({
+      exportData,
+    });
```

Comment:
`withTransactions` seems to be irrelevant technical detail there 🤔

Reply:
As we are testing the repository, we wanted to have another integrtation test for that specific method of our repository

Reply:
@AaronPerezPerez hm, steps deceived me. How about then testing it truly as an integration?

```
export = GivenTheExport()
WhenSaving(fixtures.completePieceA(export);
WhenSaving(fixtures.completePieceB(export);
ThenBothPiecesAreSaved(); // just use entity => export like in normal application and verify on "domain" object.
```

Reply:
But if we want to test the `transaction` method behaviour we should complete several pieces concurrently 🤔 . The steps of the current test are:

1. `GivenExportWithMultipleComponentsWasRequested`: Requests an export with 10 pieces
2. `WhenAllComponentsAreCompletedConcurrentlyWithTransactions`: Completes those pieces concurrently using `transaction` method
3. `WhenReadingTheSavedExportFromRepository`: Reads the export from the repo
4. `ThenAllExportComponentsShouldBeFinished`: Checks that all pieces are marked as finished

Note that the code above is outdated.